src/main/java/com/uid2/shared/attest/UidCoreClient.java

```diff
@@ -1,64 +1,52 @@
 package com.uid2.shared.attest;
 
-import com.uid2.enclave.AttestationException;
 import com.uid2.enclave.IAttestationProvider;
 import com.uid2.shared.ApplicationVersion;
 import com.uid2.shared.Const;
+import com.uid2.shared.InstantClock;
 import com.uid2.shared.Utils;
 import com.uid2.shared.cloud.*;
 import io.vertx.core.Handler;
-import io.vertx.core.json.Json;
-import io.vertx.core.json.JsonObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.crypto.Cipher;
-import javax.net.ssl.HttpsURLConnection;
 import java.io.IOException;
 import java.io.InputStream;
-import java.io.OutputStream;
-import java.net.HttpURLConnection;
-import java.net.Proxy;
-import java.net.URL;
-import java.net.URLConnection;
-import java.nio.charset.StandardCharsets;
-import java.security.*;
-import java.util.Base64;
-import java.util.Collection;
-import java.util.List;
+import java.net.*;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
 import java.util.Map;
 import java.util.concurrent.atomic.AtomicReference;
 
 public class UidCoreClient implements IUidCoreClient, DownloadCloudStorage {
     private static final Logger LOGGER = LoggerFactory.getLogger(UidCoreClient.class);
-    private final IAttestationProvider attestationProvider;
     private final ICloudStorage contentStorage;
     private final Proxy proxy;
-    private final String attestationEndpoint;
-    private final String userToken;
-    private final ApplicationVersion appVersion;
+    private String userToken;
     private final String appVersionHeader;
-    private AtomicReference<String> attestationToken;
-    private final boolean enforceHttps;
-    private static boolean useSecureParameters = true;
+    private boolean enforceHttps;
     private boolean allowContentFromLocalFileSystem = false;
-    private AtomicReference<Handler<Integer>> responseWatcher;
+    private HttpClient httpClient;
+    private AttestationTokenRetriever attestationTokenRetriever;
+    private Handler<Integer> responseWatcher;
 
-    public static UidCoreClient createNoAttest(String attestationEndpoint, String userToken, ApplicationVersion appVersion, boolean enforceHttps) {
-        return new UidCoreClient(attestationEndpoint, userToken, appVersion, CloudUtils.defaultProxy, new NoAttestationProvider(), enforceHttps);
+    public static UidCoreClient createNoAttest(String attestationEndpoint, String userToken, ApplicationVersion appVersion, boolean enforceHttps) throws IOException {
+        return new UidCoreClient(attestationEndpoint, userToken, appVersion, CloudUtils.defaultProxy, new NoAttestationProvider(), enforceHttps, null);
```

Comment:
if, instead of passing null for httpClient, you passed HttpClient.newHttpClient(), you can get rid of the test in the other constructor.